### PR TITLE
Markdown QDL, XML schema validation, and changes to QDL language.

### PR DESCRIPTION
**Pull request by smartin015**
So this one may have gotten away from me a little :P enjoy the small novel.

- Created a markdown version of QDL and converted the two existing quests to it (`<op>` doesn't yet work!)
- Created a markdown parser/converter to turn the original *.md file into *.xml file containing QDL.
- Crafted an XML schema that can be used to validate QDL in XML format
- Updated documentation for the above
- Changed `<end>` tags to `<trigger>` tags in preparation for future actions (like going back into combat)
- Small initial work into keeping game state available for quest `<op>` tags.

### CODE REVIEW

File: app/elements/quest-card.html [hunk 1]
```diff
@@ -101,6 +102,8 @@
         this._isWin = undefined;
 
         this._parser = new questParser();
+        this._parser.setGameState(this.globals);
```
Comment (smartin015) [inline app/elements/quest-card.html]: All mentions of GameState are for later work getting game state (# alive enemies, encounter tier, number of players, etc) into `<op>` codes.

File: app/quests/build/mistress_malaise.xml [hunk 2]
```diff
@@ -0,0 +1,313 @@
+<quest title="Mistress Malaise" summary="Mystery, Misfortune, and a Mistress."
```
Comment (smartin015) [inline app/quests/build/mistress_malaise.xml]: This and the Albanus quest were generated from the markdown files.

**Comment by toddmedema** [comment]
Nice work! Thoughts for testing?

**Comment by smartin015** [comment]
NONE >:{

But actually, about to begin work on a translator for the other direction. Can end-to-end test with that, then have a couple golden output files to compare against specific examples.

**Comment by toddmedema** [comment]
Gotcha

+1

**Pull request merged**